### PR TITLE
Add Dockerfile, .ci-operator.yaml for upstream ci

### PR DESCRIPTION
**Pull request by Madhan-SWE**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adding Docker file and .ci-operator.yaml files for upstream ci.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by Madhan-SWE** [comment]
@mkumatag , can you review this?

File: .ci-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+build_root_image:
+  name: release
+  namespace: openshift
+  tag: rhel-8-release-golang-1.17-openshift-4.11
```
Comment (mkumatag) [inline .ci-operator.yaml:4]: how about using 1.18 here because the code needs 1.18 right?

File: Dockerfile.openshift [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.11 AS builder
```
Comment (mkumatag) [inline Dockerfile.openshift:1]:
```suggestion
FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.12 AS builder
```

check if this image exist and change accordingly.

File: Dockerfile.openshift [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+WORKDIR /go/src/github.com/openshift/ibm-powervs-block-csi-driver
+COPY . .
+RUN make driver
+
+FROM registry.ci.openshift.org/ocp/4.11:base
```
Comment (mkumatag) [inline Dockerfile.openshift:6]:
```suggestion
FROM registry.ci.openshift.org/ocp/4.12:base
```

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: .ci-operator.yaml (hunk 1 above)
Comment (mkumatag) [reply .ci-operator.yaml:4]: go.mod file can have a lower version anyways but we can stick to golang 1.17 as of now.

**Pull request merged**